pkg/ovs/openflow/ofctrl_bridge.go

```diff
@@ -240,6 +242,40 @@ func (b *OFBridge) DeleteMeterAll() error {
 	return nil
 }
 
+func (b *OFBridge) GetMeterStats(handleMeterStatsReply func(packetCounts map[int]int64)) error {
+	const OFPM_ALL = 0xffffffff // Represents all meters
+	mpMeterStatsReq := openflow15.NewMpRequest(openflow15.MultipartType_MeterStats)
+	meterMPReq := openflow15.NewMeterMultipartRequest(OFPM_ALL)
+	mpMeterStatsReq.Body = append(mpMeterStatsReq.Body, meterMPReq)
+	ch := make(chan *openflow15.MultipartReply, 1)
+	b.mpReplyChsMutex.Lock()
+	defer b.mpReplyChsMutex.Unlock()
+	b.multipartReplyChs[mpMeterStatsReq.Xid] = ch
+	go func() {
+		defer func() {
+			b.mpReplyChsMutex.Lock()
```

Comment:
I think you can still us a `defer` statement here for the unlock

It does feel like the code would be a bit cleaner with 2 helper function:

```go
func (b *OFBridge) registerMultipartReplyCh(xid int, ch chan *openflow15.MultipartReply) {

}

func (b *OFBridge) unregisterMultipartReplyCh(xid int) {

}
```

what do you think?

Reply:
It is great to me. Thanks for the suggestion, I will update this part.